src/set_theory/game/pgame.lean

```diff
@@ -1065,4 +1069,72 @@ begin
   exact zero_lt_one,
 end
 
+/-- The birthday of a pre-game is inductively defined as the least strict upper bound of the
+birthdays of its left and right games. It may be thought as the "step" in which a certain game is
+constructed. -/
+noncomputable def birthday : pgame.{u} → ordinal.{u}
+| ⟨xl, xr, xL, xR⟩ :=
+    max (ordinal.lsub.{u u} $ λ i, birthday (xL i)) (ordinal.lsub.{u u} $ λ i, birthday (xR i))
+
+@[simp] theorem birthday_mk (xl xr xL xR) : birthday (mk xl xr xL xR) = max
+  (ordinal.lsub.{u u} (λ i, birthday (xL i)))
+  (ordinal.lsub.{u u} (λ i, birthday (xR i))) :=
+by rw birthday
+
+theorem birthday_def (x : pgame) : birthday x = max
+  (ordinal.lsub.{u u} (λ i, birthday (x.move_left i)))
+  (ordinal.lsub.{u u} (λ i, birthday (x.move_right i))) :=
+by { cases x, apply birthday_mk }
+
+theorem left_move_birthday_lt {x : pgame} (i : x.left_moves) :
+  (x.move_left i).birthday < x.birthday :=
+by { cases x, rw birthday_mk, exact lt_max_of_lt_left (ordinal.lt_lsub _ i) }
+
+theorem right_move_birthday_lt {x : pgame} (i : x.right_moves) :
+  (x.move_right i).birthday < x.birthday :=
+by { cases x, rw birthday_mk, exact lt_max_of_lt_right (ordinal.lt_lsub _ i) }
+
+theorem lt_birthday_iff {x : pgame} {o : ordinal} : o < x.birthday ↔
+  (∃ i : x.left_moves, o ≤ (x.move_left i).birthday) ∨
+  (∃ i : x.right_moves, o ≤ (x.move_right i).birthday) :=
+begin
+  split,
+  { rw birthday_def,
+    intro h,
+    cases lt_max_iff.1 h with h' h',
+    { left,
+      rwa ordinal.lt_lsub_iff at h' },
+    { right,
+      rwa ordinal.lt_lsub_iff at h' } },
+  { rintro (⟨i, hi⟩ | ⟨i, hi⟩),
+    { exact hi.trans_lt (left_move_birthday_lt i) },
+    { exact hi.trans_lt (right_move_birthday_lt i) } }
+end
+
+theorem relabelling.birthday_congr : ∀ {x y : pgame.{u}}, relabelling x y → birthday x = birthday y
+| ⟨xl, xr, xL, xR⟩ ⟨yl, yr, yL, yR⟩ ⟨L, R, hL, hR⟩ := begin
+  rw [birthday, birthday],
+  congr' 1,
+  all_goals
+  { apply ordinal.lsub_eq_of_range_eq.{u u u},
+    ext i,
+    split },
+  { rintro ⟨j, rfl⟩,
+    exact ⟨L j, (relabelling.birthday_congr (hL j)).symm⟩ },
+  { rintro ⟨j, rfl⟩,
+    refine ⟨L.symm j, relabelling.birthday_congr _⟩,
+    convert hL (L.symm j),
+    rw L.apply_symm_apply },
+  { rintro ⟨j, rfl⟩,
+    refine ⟨R j, (relabelling.birthday_congr _).symm⟩,
+    convert hR (R j),
+    rw R.symm_apply_apply },
+  { rintro ⟨j, rfl⟩,
+    exact ⟨R.symm j, relabelling.birthday_congr (hR j)⟩ }
+end
+using_well_founded { dec_tac := pgame_wf_tac }
+
+@[simp] theorem birthday_zero : birthday 0 = 0 :=
```

Comment:
Is there anything interesting to state about 1 or `birthday -x`?

Reply:
There are lots of theorems here about birthdays of natural numbers, dyadics, etc, I think for this PR just the zero-th theorem is okay. :-)

Reply:
Yeah, there's a ton of theorems, but I'd rather just start with the basic theorems and then build up on it.

Reply:
I'm just thinking about whether there are any other ones that have a proof of `rfl` or `cases x, refl`, as those would be easy to add in this PR

Reply:
I couldn't think of any. Even proving that `birthday 1 = 1` is nontrivial (made particularly annoying by the fact `0 + 1` is only a relabeling and not equal to `1`). In fact, I'd rather wait for the `ordinal.to_pgame` PR to then directly prove `birthday o.to_pgame = o`, and use that and a few relabelings to get the theorems for naturals.